src/Components/Aspire.Milvus.Client/AspireMilvusExtensions.cs

```diff
@@ -0,0 +1,96 @@
+    /// <remarks>Reads the configuration from "Aspire:Milvus:Client" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddMilvusClient(
+        this IHostApplicationBuilder builder,
+        string connectionName,
+        Action<MilvusClientSettings>? configureSettings = null)
+    {
+        AddMilvus(builder, DefaultConfigSectionName, configureSettings, connectionName, serviceKey: null);
+    }
+
+    /// <summary>
+    /// Registers <see cref="MilvusClient" /> as a keyed singleton for the given <paramref name="name" /> in the services provided by the <paramref name="builder"/>.
+    /// Configures logging for the <see cref="MilvusClient" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IHostApplicationBuilder" /> to read config from and add services to.</param>
+    /// <param name="name">The connection name to use to find a connection string.</param>
+    /// <param name="configureSettings">An optional method that can be used for customizing the <see cref="MilvusClientSettings"/>. It's invoked after the settings are read from the configuration.</param>
+    /// <remarks>Reads the configuration from "Aspire:Milvus:Client" section.</remarks>
+    /// <exception cref="InvalidOperationException">If required ConnectionString is not provided in configuration section</exception>
+    public static void AddKeyedMilvusClient(
+        this IHostApplicationBuilder builder,
+        string name,
+        Action<MilvusClientSettings>? configureSettings = null)
+    {
+        AddMilvus(builder, $"{DefaultConfigSectionName}:{name}", configureSettings, connectionName: name, serviceKey: name);
+    }
+
+    private static void AddMilvus(
+        this IHostApplicationBuilder builder,
+        string configurationSectionName,
+        Action<MilvusClientSettings>? configureSettings,
+        string connectionName,
+        string? serviceKey)
+    {
+        ArgumentNullException.ThrowIfNull(builder);
+
+        var settings = new MilvusClientSettings();
+        builder.Configuration.GetSection(configurationSectionName).Bind(settings);
+
+        if (builder.Configuration.GetConnectionString(connectionName) is string connectionString)
+        {
+            settings.ParseConnectionString(connectionString);
+        }
+
+        configureSettings?.Invoke(settings);
+
+        if (serviceKey is null)
+        {
+            builder.Services.AddSingleton(ConfigureMilvus);
+        }
+        else
+        {
+            builder.Services.AddKeyedSingleton(serviceKey, (sp, key) => ConfigureMilvus(sp));
+        }
+
+        MilvusClient ConfigureMilvus(IServiceProvider serviceProvider)
+        {
+            if (settings.Endpoint is not null)
+            {
+                return new MilvusClient(settings.Endpoint, apiKey: settings.Key!, loggerFactory: serviceProvider.GetRequiredService<ILoggerFactory>());
```

Comment:
Should we also be checking for the Key being null, and throwing a meaningful exception if it is null?